gdbinit.py

```diff
@@ -16,8 +18,46 @@
     _start_time = time.time()
     _profiler.enable()
 
+
 # Get virtualenv's site-packages path
 venv_path = os.environ.get("PWNDBG_VENV_PATH")
+
+
+def calculate_hash(file_path):
+    return hashlib.file_digest(open(file_path, "rb"), "sha256").hexdigest()
+
+
+def run_poetry_install(dev=False):
+    command = ["poetry", "install"]
+    if dev:
+        command = ["poetry", "install", "--with", "dev"]
+    subprocess.run(command, check=True)
+
+
+PWNDBG_VENV_PATH = os.getenv("PWNDBG_VENV_PATH")
+POETRY_LOCK_PATH = os.path.join(os.path.dirname(__file__), "poetry.lock")
+POETRY_LOCK_HASH_PATH = os.path.join(PWNDBG_VENV_PATH, "poetry.lock.hash")
+DEV_MARKER_PATH = os.path.join(PWNDBG_VENV_PATH, "dev.marker")
+
+# verify virtual environment path exists
+os.makedirs(PWNDBG_VENV_PATH, exist_ok=True)
```

Comment:
I think we can get rid of this if we move this code below the `if not os.path.exists(venv_path):` check at line 71.